### PR TITLE
Refs #33342 -- Deprecated ExclusionConstraint.opclasses.

### PR DESCRIPTION
**Pull request by hannseman**
Ticket: https://code.djangoproject.com/ticket/33342

### CODE REVIEW

**Comment by felixxm** [comment]
@hannseman Thanks :+1: I pushed the first commit with small edits to the #15235. Let's handle deprecation separately.

**Comment by felixxm** [comment]
@hannseman Please rebase from the `main` branch.

**Comment by hannseman** [comment]
Done, Merry Christmas! 🎄

File: tests/postgres_tests/test_constraints.py [hunk 1]
```diff
@@ -856,6 +878,11 @@ def test_spgist_include_not_supported(self):
                 with self.assertRaisesMessage(NotSupportedError, msg):
                     editor.add_constraint(RangesModel, constraint)
 
+    # RemovedInDjango50Warning: When the deprecation ends, remove the
+    # @ignore_warnings and opclasses=['range_ops']. Replace the
+    # expressions argument with:
+    # expressions=[(OpClass('ints', name='range_ops'), RangeOperators.ADJACENT_TO)]
+    @ignore_warnings(category=RemovedInDjango50Warning)
```
Comment (felixxm) [inline tests/postgres_tests/test_constraints.py:1001]: We should have both versions covered by tests, so I moved test methods with `opclasses` to the new class `ExclusionConstraintOpclassesDepracationTests`.

**Review by felixxm — COMMENTED** [review]
@hannseman Thanks :+1: I pushed small edits to docs and added `ExclusionConstraintOpclassesDepracationTests`.

**Pull request merged**